x-pack/plugins/maps/public/shared/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AbstractVectorSource } from '../vector_source';
+import React from 'react';
+import { GEOJSON_FILE } from '../../../../../common/constants';
+import { ClientFileCreateSourceEditor } from './create_client_file_source_editor';
+import { ESSearchSource } from '../es_search_source';
+import uuid from 'uuid/v4';
+import _ from 'lodash';
+
+export class GeojsonFileSource extends AbstractVectorSource {
+
+  static type = GEOJSON_FILE;
+  static title = 'Import GeoJSON Vector File';
+  static description = 'Upload a GeoJSON file & Index in Elasticsearch';
```

Comment:
Would talk to @gchaps to fine tune some of this wording and formatting

Reply:
For the title, use sentence case:

Import GeoJSON vector file

A few comments on the description:

- It uses "upload" where the title uses "import".  Can this be consistent?
- It says close the the same thing as the title.  Is it possible to give more information?
- It should use sentence case and ending punctuation:

Upload a GeoJSON file and index in Elasticsearch.

Reply:
I ended up not adding a period at the end of the description to match the other cards. If we rethink that, we should just update them all in a separate PR. All other updates made. I couldn't think of much more to the description beyond what we've said that doesn't sound like fluff. We can certainly iterate on this though as needed!